clang/test/CodeGenSYCL/sycl-external-attr.cpp

```diff
@@ -0,0 +1,77 @@
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
+
+// This test code generation when sycl_external attribute is used
+
+// Function defined and not used - symbols emitted
+[[clang::sycl_external]] int square(int x) { return x*x; }
+// CHECK: define dso_local spir_func noundef i32 @_Z6squarei
+
+// Function defined and used - symbols emitted
+[[clang::sycl_external]] int squareUsed(int x) { return x*x; }
+// CHECK: define dso_local spir_func noundef i32 @_Z10squareUsedi
+
+// Constexpr function defined and not used - symbols emitted
```

Comment:
Let's note this as a FIXME since there is no need to emit an unused inline function.
```suggestion
// FIXME: Constexpr function defined and not used - no definition emitted
```

Reply:
I was under the opposite impression per one of the older review comments but I can't find it at the moment. No related wording in the spec either.

Reply:
Functions declared `constexpr` are implicitly `inline` and are therefore available with the same definition in every translation unit that declares them (under penalty of the ODR). We therefore shouldn't have to emit an unused one.